internal/service/http/auth.go

```diff
@@ -0,0 +1,75 @@
+package http
+
+import (
+	"crypto/sha256"
+	"crypto/subtle"
+	"errors"
+	"net/http"
+	"strings"
+
+	"github.com/grafana/alloy/syntax/alloytypes"
+)
+
+type AuthArguments struct {
+	Basic *BasicAuthArguments `alloy:"basic,block,optional"`
+	// Filter is used to apply authentication to matching api endpoints.
+	Filter []string `alloy:"filter,attr,optional"`
```

Comment:
In https://github.com/grafana/alloy/issues/637#issuecomment-2602834802 I noticed that this kind of syntax was used for filter: `['/debug/pprof', '/component/*']`

But I figured it was easier to implement as prefix matching instead of having some kind of wildcard symbol

Reply:
Yes, this is probably sufficient and will be easier for most people to configure than regex/wildcards.